src/core/radio/trel_interface.hpp

```diff
@@ -230,10 +230,11 @@ class Interface : public InstanceLocator
 
 private:
     static constexpr uint16_t kPeerTableExtraEntries = 32;
-    static constexpr uint16_t kPeerTableSize         = Mle::kMaxRouters + Mle::kMaxChildren + kPeerTableExtraEntries;
-
-    static const char kTxtRecordExtAddressKey[];
-    static const char kTxtRecordExtPanIdKey[];
+    static constexpr uint16_t kPeerTableSize         = OPENTHREAD_CONFIG_TREL_PEER_TABLE_SIZE
```

Comment:
The direct use of MACRO definition in `? :` (and `constexpr` init) may result in warning in some toolchains.
Better way is to use the following format:
```c++
#if OPENTHREAD_COFNIG_TREL_PEER_TABLE_SIZE != 0
      static constexpr uint16_t kPeerTableSize = OPENTHREAD_COFNIG_TREL_PEER_TABLE_SIZE;
#else
      static constexpr uint16_t kPeerTableSize = ...
#endif
```

Reply:
Done.